dataclients/kubernetes/fixtures_test.go

```diff
@@ -0,0 +1,269 @@
+				fixtures.kube = filepath.Join(dir, n)
+			case ".default-filters":
+				fixtures.defaultFilters = filepath.Join(dir, n)
+			case ".error":
+				fixtures.error = filepath.Join(dir, n)
+			case ".log":
+				fixtures.log = filepath.Join(dir, n)
+			}
+
+			fs = fs[1:]
+		}
+
+		test(fixtures)
+		fixtures = empty
+	}
+}
+
+func matchOutput(matchFile, output string) error {
+	b, err := ioutil.ReadFile(matchFile)
+	if err != nil {
+		return err
+	}
+
+	exps := strings.Split(string(b), "\n")
+	lines := strings.Split(output, "\n")
+	for _, e := range exps {
+		rx := regexp.MustCompile(e)
+
+		var found bool
+		for _, l := range lines {
+			if rx.MatchString(l) {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			return fmt.Errorf("not matched: '%s'", e)
+		}
+	}
+
+	return nil
+}
+
+func testFixture(t *testing.T, f fixtureSet) {
+	var resources []io.Reader
+	if f.resources != "" {
+		r, err := os.Open(f.resources)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		defer r.Close()
+		resources = append(resources, r)
+	}
+
+	var apiOptions testAPIOptions
+	if f.api != "" {
+		a, err := os.Open(f.api)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		defer a.Close()
+		apiOptions, err = readAPIOptions(a)
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+
+	a, err := newAPI(apiOptions, resources...)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	s := httptest.NewServer(a)
+	defer s.Close()
+
+	var logBuf bytes.Buffer
+	// TODO: we should refactor the package to not use the global logger
+	log.SetOutput(&logBuf)
+	defer log.SetOutput(os.Stderr)
+	defer func() {
+		l := logBuf.String()
+		if l != "" {
+			t.Log("Captured logs:")
+			t.Log(strings.TrimSpace(l))
+		}
+	}()
+
+	var o Options
+	if f.kube != "" {
+		ko, err := os.Open(f.kube)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		defer ko.Close()
+		b, err := ioutil.ReadAll(ko)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		var kop kubeOptionsParser
+		if err := yaml.Unmarshal(b, &kop); err != nil {
+			t.Fatal(err)
+		}
+
+		o.KubernetesEnableEastWest = kop.EastWest
+		o.KubernetesEastWestDomain = kop.EastWestDomain
+		o.ProvideHTTPSRedirect = kop.HTTPSRedirect
+		o.HTTPSRedirectCode = kop.HTTPSRedirectCode
+	}
+
+	o.KubernetesURL = s.URL
+	o.DefaultFiltersDir = f.defaultFilters
+	c, err := New(o)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer c.Close()
+
+	routes, err := c.LoadAll()
+	if f.eskip != "" {
+		eskp, err := os.Open(f.eskip)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		defer eskp.Close()
+		b, err := ioutil.ReadAll(eskp)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		expectedRoutes, err := eskip.Parse(string(b))
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if !eskip.EqLists(routes, expectedRoutes) {
+			t.Error("Failed to convert the resources to the right routes.")
+			t.Logf("routes: %d, expected: %d", len(routes), len(expectedRoutes))
+			t.Logf("got:\n%s", eskip.String(eskip.CanonicalList(routes)...))
+			t.Logf("expected:\n%s", eskip.String(eskip.CanonicalList(expectedRoutes)...))
```

Comment:
not sure if `cmp.Diff(eskip.String(eskip.CanonicalList(routes)...), eskip.String(eskip.CanonicalList(expectedRoutes)...))` would not help to show the error better

Reply:
added diff, but I left the current one, too, because I found it useful to have an overview while working on these tests, as sometimes the diff was too big.